### PR TITLE
Add support for retrieving request executionDuration.

### LINKED ISSUE #2986: Add "took" attribute to SearchHits/ReactiveSearchHits
I am working on a Spring API that integrates with Elasticsearch, and my current task involves migrating from the **Elasticsearch Java API Client** to **Spring Data Elasticsearch**. While doing so, I noticed that the response body in the Elasticsearch Java API Client includes a valuable attribute called **"took"**, which shows the time (in milliseconds) it took to complete the request. However, this attribute is not accessible in the **SearchHits** (or **ReactiveSearchHits**) body in Spring Data Elasticsearch.

This feature is important for performance monitoring and debugging, as it allows developers to measure the responsiveness of Elasticsearch queries.

I propose adding support for retrieving the **"took"** attribute from **SearchHits** objects in **Spring Data Elasticsearch**, ensuring **feature parity** with the **Elasticsearch Java API Client**. This enhancement would improve the developer experience by providing essential timing information for queries.

I would be happy to contribute to this feature by submitting a PR if accepted. Thank you!

### PR DESCRIPTION
**Pull request by ELHARROUGUI**
Please let me know if I missed anything.

- [x] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [x] **There is a ticket in the bug tracker for the project in our [issue tracker](https://github.com/spring-projects/spring-data-elasticsearch/issues)**. Add the issue number to the _Closes #issue-number_ line below
- [x] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

Closes #2986

### CODE REVIEW

File: src/test/java/org/springframework/data/elasticsearch/core/ElasticsearchIntegrationTests.java [hunk 1]
```diff
@@ -1872,6 +1874,7 @@ public void shouldReturnScores() {
 				IndexCoordinates.of(indexNameProvider.indexName()));
 
 		assertThat(searchHits.getMaxScore()).isGreaterThan(0f);
+		assertThat(searchHits.getExecutionDuration().toMillis()).isGreaterThan(0);
```
Comment (sothawo) [inline src/test/java/org/springframework/data/elasticsearch/core/ElasticsearchIntegrationTests.java:1877]: Please add "#2986" to the comment in line 1861 (`@Test // DATAES-462`)

**Review by sothawo — CHANGES_REQUESTED** [review]
thanks for that PR; 2 changes to do

File: src/test/java/org/springframework/data/elasticsearch/core/ElasticsearchIntegrationTests.java (hunk 1 above)
Comment (ELHARROUGUI) [reply src/test/java/org/springframework/data/elasticsearch/core/ElasticsearchIntegrationTests.java:1877]: Done.

File: pom.xml [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
 	<groupId>org.springframework.data</groupId>
 	<artifactId>spring-data-elasticsearch</artifactId>
-	<version>5.4.0-SNAPSHOT</version>
+	<version>5.4.0-2986-SNAPSHOT</version>
```
[the remark this answers is not in the document]
Comment (ELHARROUGUI) [reply pom.xml]: Done. I used this [script ](https://github.com/spring-projects/spring-data-build/blob/main/etc/scripts/new-issue-branch) to create the issue branch so the script made that change too.

**Review by sothawo — APPROVED** [review]

**Pull request merged**